tests/request_parsing_tests.rs

```diff
@@ -0,0 +1,187 @@
+use std::borrow::Cow;
+
+use x11rb::errors::ParseError;
+use x11rb::x11_utils::RequestHeader;
+
+#[test]
+fn test_bad_request_header_opcode() {
+    use x11rb::protocol::xproto::QueryExtensionRequest;
+    let header = RequestHeader {
+        major_opcode: 1,
+        minor_opcode: 2,
+        remaining_length: 0,
+    };
+    let body = &[];
+    assert_eq!(
+        QueryExtensionRequest::try_parse_request(header, body),
+        Err(ParseError::ParseError)
+    );
+}
+
+#[test]
+fn test_bad_request_header_length() {
+    use x11rb::protocol::xproto::CreateWindowRequest;
+    let header = RequestHeader {
+        major_opcode: 1,
+        minor_opcode: 0,
+        remaining_length: 42,
+    };
+    let body = &[];
+    assert_eq!(
+        CreateWindowRequest::try_parse_request(header, body),
+        Err(ParseError::ParseError)
+    );
+}
+
+#[test]
+fn test_create_window1() {
+    use x11rb::protocol::xproto::{CreateWindowAux, CreateWindowRequest, Gravity, WindowClass};
+    let header = RequestHeader {
+        major_opcode: 1,
+        minor_opcode: 0x18,
+        remaining_length: 10,
+    };
+    let body = &[
+        0x1, 0x0, 0xc0, 0x5, 0x13, 0x5, 0x0, 0x0, 0x7b, 0x4, 0x34, 0x1, 0xf5, 0x3, 0x3b, 0x3, 0x0,
+        0x0, 0x1, 0x0, 0x47, 0x0, 0x0, 0x0, 0x1a, 0x0, 0x0, 0x0, 0xf0, 0xf1, 0xf2, 0xff, 0x0, 0x0,
+        0x0, 0x0, 0x1, 0x0, 0x0, 0x0,
+    ];
```

Comment:
I believe this is little-endian only, isn't it?

Reply:
Oh right, I totally missed that. Endianess is why `tests/parsing_tests.rs` needs so many lines to construct the setup bytes.

Thanks for catching this.

Reply:
This is fixed now.